docs/testplan/BGP-Suppress-FIB-Pending-test-plan.md

```diff
@@ -0,0 +1,173 @@
+```
+9. Make sure announced BGP routes are not in __queued__ state in the DUT routing table
+10. Make sure the routes are programmed in FIB by checking __offloaded__ flag in the DUT routing table
+```
+show ip route 1.1.1.0/24 json
+
+{
+    "1.1.1.0/24": [
+        {
+            "destSelected": true,
+            "distance": 20,
+            "installed": true,
+            "installedNexthopGroupId": 277,
+            "internalFlags": 264,
+            "internalNextHopActiveNum": 1,
+            "internalNextHopNum": 1,
+            "internalStatus": 80,
+            "metric": 0,
+            "nexthopGroupId": 277,
+            "nexthops": [
+                {
+                    "active": true,
+                    "afi": "ipv4",
+                    "fib": true,
+                    "flags": 3,
+                    "interfaceIndex": 17,
+                    "interfaceName": "PortChannel1021",
+                    "ip": "10.0.0.65",
+                    "weight": 1
+                }
+            ],
+            "offloaded": true,
+            "prefix": "1.1.1.0/24",
+            "prefixLen": 24,
+            "protocol": "bgp",
+            "selected": true,
+            "table": 254,
+            "uptime": "00:08:08",
+            "vrfId": 0,
+            "vrfName": "default"
+        }
+    ]
+}
+```
+11. Verify the routes are announced to T2 peer
+12. Send traffic matching the prefixes and verify packets are forwarded to T0 VM
+13. This test should cover __default__ and __user defined vrf__
+
+### Test case # 2 - BGPv6 route suppress test
+1. Enable BGP suppress-fib-pending function at DUT
+2. Save configuration and execute one action randomly choosen from(__reboot__/__config reload__/__fast-reboot__/__warm-reboot__)
+3. Suspend orchagent process to simulate a delay
+4. Announce BGP ipv6 prefixes to DUT from T0 VM by exabgp
+5. Make sure announced BGP routes are in __queued__ state in the DUT routing table
+6. Verify the routes are not announced to T2 VM peer
+7. Send traffic matching the prefixes and verify packets are not forwarded to T0 VM
+8. Restore orchagent process
+9. Make sure announced BGP routes are not in __queued__ state in the DUT routing table
+10. Make sure the routes are programmed in FIB by checking __offloaded__ flag in the DUT routing table
+11. Verify the routes are announced to T2 peer
+12. Send traffic matching the prefixes and verify packets are forwarded to T0 VM
+13. This test should cover __default__ and __user defined vrf__
+
+### Test case # 3 - Test BGP route without suppress
+1. No BGP suppress-fib-pending function configured at DUT
+2. Suspend orchagent process to simulate a delay
+3. Announce BGP prefixes to DUT from T0 VM by exabgp
+4. Make sure announced BGP routes are not in __queued__ state in the DUT routing table
+5. Verify the BGP routes are announced to T2 peer
+6. Restore orchagent process
+7. Make sure the routes are programmed in FIB by checking __offloaded__ flag in the DUT routing table
```

Comment:
if no BGP suppress fib pending configured, offload flag would be set directly, right, should we check it before step 5? @stepanblyschak

Reply:
> if no BGP suppress fib pending configured, offload flag would be set directly, right, should we check it before step 5? @stepanblyschak

This step is to make sure, no queued route should be exist in this scenario.

Reply:
> @StormLiangMS The fact that it is not in **queued** state implies it is **offloaded**.

Yes, you are right.